src/main/java/sg/dex/crypto/Hash.java

```diff
@@ -123,4 +127,34 @@ public static String sha3_256String(String string) {
         return Hex.toString(sha3_256(string));
     }
 
+    /**
+     * This method is used to compute the complete hash computation based on input stream passed.
+     * A MessageDigest object starts out initialized.
+     * The data is processed through it using the update methods.
+     * At any point reset can be called to reset the digest.
+     * Once all the data to be updated has been updated, one of the digest
+     * methods should be called to complete the hash computation.
+     *
+     * @param inputStream
+     * @return
+     */
+    public static String checkSum(InputStream inputStream) throws DigestException {
+        String checksum = null;
+        try {
+
+            SHA3.DigestSHA3 md = new SHA3.Digest256();
+            //Using MessageDigest update() method to provide input
+            byte[] buffer = new byte[8192];
+            int numOfBytesRead;
+            while ((numOfBytesRead = inputStream.read(buffer)) > 0) {
+                md.update(buffer, 0, numOfBytesRead);
+            }
+            byte[] hash = md.digest();
+            checksum = new BigInteger(1, hash).toString(16); //don't use this, truncates leading zero
```

Comment:
Don't use BigInteger here. Use a static utility method to convert byte[] to hex String, I think one exists already.

Reply:
done.